win32/src/async_socket_win32.c

```diff
@@ -726,6 +739,10 @@ int async_socket_receive_async(ASYNC_SOCKET_HANDLE async_socket, ASYNC_SOCKET_BU
                             /* Codes_SRS_ASYNC_SOCKET_WIN32_01_081: [ An asynchronous IO shall be started by calling StartThreadpoolIo. ]*/
                             StartThreadpoolIo(async_socket->tp_io);
 
+#ifdef ENABLE_SOCKET_LOGGING
+                            LogVerbose("Starting receive at %lf", timer_global_get_elapsed_us());
+#endif
+
                             /* Codes_SRS_ASYNC_SOCKET_WIN32_01_082: [ A receive shall be started for the WSABUF array associated with the context calling WSARecv and passing to it the OVERLAPPED structure with the event that was just created, dwFlags set to 0, lpNumberOfBytesSent set to NULL and lpCompletionRoutine set to NULL. ]*/
                             wsa_receive_result = WSARecv((SOCKET)async_socket->socket_handle, receive_context->wsa_buffers, buffer_count, NULL, &flags, &receive_context->overlapped, NULL);

```

Comment:
>wsa_receive_result [](http://example.com/codeflow?start=28&length=18)

Can a receive be "synchronous"? seems not by the way the code is written.

Reply:
Not right now. I think we have a task somewhere that we might want to do that, but right now everything is async path.

---
In reply to: [654743180](https://github.com/Azure/c-pal/pull/95#discussion_r654743180) [](ancestors = 654743180)